src/components/AccountDetailsV2/TransactionBody.tsx

```diff
@@ -0,0 +1,171 @@
+import { Trans } from '@lingui/macro'
+import { Fraction, TradeType } from '@uniswap/sdk-core'
+import JSBI from 'jsbi'
+import {
+  AddLiquidityV3PoolTransactionInfo,
+  ExactInputSwapTransactionInfo,
+  ExactOutputSwapTransactionInfo,
+  RemoveLiquidityV3TransactionInfo,
+  TransactionInfo,
+  TransactionType,
+} from 'state/transactions/types'
+import styled from 'styled-components/macro'
+
+import { useCurrency } from '../../hooks/Tokens'
+import { TransactionState } from './index'
+
+const HighlightText = styled.span`
+  color: ${({ theme }) => theme.textPrimary};
+  font-weight: 600;
+`
+
+const formatAmount = (amountRaw: string, decimals: number, sigFigs: number): string =>
+  new Fraction(amountRaw, JSBI.exponentiate(JSBI.BigInt(10), JSBI.BigInt(decimals))).toSignificant(sigFigs)
+
+const getFailedText = (transactionState: TransactionState) =>
```

Comment:
any time text fails, it takes on the form of

 Swap [TOKEN] for [TOKEN] failed
 
It was a lot easier for me to wrap this in a function and append it at the end of every  transaction